### PR TITLE
Base64Util (#188 continued)

### PR DESCRIPTION
**Pull request by RikkiGibson**

### CODE REVIEW

File: test/azure/pom.xml [hunk 1]
```diff
@@ -65,6 +65,13 @@
       <artifactId>slf4j-simple</artifactId>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.microsoft.azure</groupId>
+      <artifactId>codegen-tests</artifactId>
+      <version>1.0.0-SNAPSHOT</version>
+      <type>test-jar</type>
```
Comment (RikkiGibson) [inline test/azure/pom.xml]: This lets us share RunServerListener with the other modules even though it lives under src/test/java.

**Comment by RikkiGibson** [comment]
Going to release today but would like your review after the fact anyway @daschult

File: test/vanilla/src/test/java/fixtures/server/RunServerListener.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+
+public class RunServerListener extends RunListener {
+    private Process serverProcess;
+    private Thread readerThread;
+
+    @Override
+    public void testRunStarted(Description description) throws Exception {
+        ProcessBuilder pb = new ProcessBuilder("node", "../../node_modules/@microsoft.azure/autorest.testserver")
+                .redirectErrorStream(true);
```
Comment (daschult) [inline test/vanilla/src/test/java/fixtures/server/RunServerListener.java]: Just to make sure, do we need to redirect the OutputStream as well?

File: test/vanilla/src/test/java/fixtures/server/RunServerListener.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                Path outPath = Paths.get("testserver.log");
+                Files.deleteIfExists(outPath);
+
+                BufferedReader reader = new BufferedReader(new InputStreamReader(serverProcess.getInputStream()));
+                BufferedWriter writer = Files.newBufferedWriter(outPath);
+
+                String line = reader.readLine();
+                writer.append(line);
+                writer.newLine();
+                LoggerFactory.getLogger(getClass()).info(line);
```
Comment (daschult) [inline test/vanilla/src/test/java/fixtures/server/RunServerListener.java]: Why does this only happen for the first line?

**Pull request merged**